index.js

```diff
@@ -1,76 +1,119 @@
+  })
+  .option("dry-run", {
+    type: "boolean",
+    default: false,
+    describe: "Script will not submit HTTP requests, if this flag is enabled."
+  })
+  .demandOption(["jiraUrl", "confUrl", "csv", "appId"])
+  .argv;
 
 const ISSUE_KEY_REGEX = /(?<=key&quot;&gt;)[A-Z]+-[0-9]+(?=&lt;)/g;
 
-// TODO Add feature for dry run - don't make HTTP requests
+const jiraEndpoint = argv['jira-url'];
+const confluenceEndpoint = argv['conf-url'];
+const csvFilename = argv['csv'];
+const appId = argv['app-id'];
+const usernameAndPassword = argv['jira-auth'];
+const isDryRun = argv['dry-run'];
 
-const jiraEndpoint = argv.jiraUrl;
-const confluenceEndpoint = argv.confUrl;
-const csvFilename = argv.csv;
-const appId = argv.appId;
+const requests = [];
+let createdLinksCount = 0;
+let preexistingLinksCount = 0;
+let errCount = 0;
 
-// Send HTTP requests to Jira
 const processRow = row => {
 
-    // Q: What about versions? Do we need to update on all versions?
+  const pageId = row["Page ID"];
+  const title = row["Page Title"];
+  const metadata = row["text_val"];
 
-    const pageId = row["Page ID"];
-    const title = row["Page Title"];
-    const metadata = row["text_val"];
+  const issueKeys = metadata.match(ISSUE_KEY_REGEX);
+  log(INFO, `[PAGE-ID: ${pageId}] Keys: ${issueKeys}`);
 
-    const issueKeys = metadata.match(ISSUE_KEY_REGEX);
-    console.log("Keys: ", issueKeys);
+  const body = {
+    globalId: `appId=${appId}&pageId=${pageId}`,
+    application: {
+      type: "com.atlassian.confluence",
+      name: "Confluence"
+    },
+    relationship: "mentioned in",
+    object: {
+      url: `${confluenceEndpoint}/pages/viewpage.action?pageId=${pageId}`,
+      title
+    }
+  };
 
-    const body = {
-        globalId: `appId=${appId}&pageId=${pageId}`,
-        application: {
-            type: "com.atlassian.confluence",
-            name: "Confluence"
-        },
-        relationship: "mentioned in",
-        object: {
-            url: `${confluenceEndpoint}/pages/viewpage.action?pageId=${pageId}`,
-            title
-        }
-    };
+  issueKeys.forEach(issueKey => {
+    const requestUrl = `${jiraEndpoint}/rest/api/latest/issue/${issueKey}/remotelink`;
 
-    console.log(body);
+    if (isDryRun) {
+      log(INFO, `[DRY-RUN] Will update ${requestUrl}`);
+      return;
+    }
 
-    issueKeys.forEach(issueKey => {
-        const requestUrl = `${jiraEndpoint}/rest/api/latest/issue/${issueKey}/remotelink`;
-        console.log(requestUrl);
-    });
+    const authBuffer = Buffer.from(usernameAndPassword, 'utf-8');
+    requests.push(
+      axios.post(requestUrl, body, {
+        headers: {
+          'Authorization': `Basic ${authBuffer.toString('base64')}`,
+          'Content-Type': 'application/json'
+        }
+      })
+        .then(result => {
+          result.status === 201 ? createdLinksCount++ : preexistingLinksCount++;
```

Comment:
Suggest to fix the indentation here

Reply:
Do you mean the `.then` block? I think it's commonly indented. In any case, I am using intelliJ's formatter and it's making it this way. Let me know if you were actually referring to something else.

Reply:
Yeah, I'm looking at line 95 actually, the indentation looks out of place to me

Reply:
We can compare with line 99-100 which is what I usually expect. Anyway no strong opinion on this one, it's just indentation

Reply:
Actually, line 100 has the same indentation as line 96.
![Screenshot 2021-03-25 at 12 09 41](https://user-images.githubusercontent.com/44353593/112417668-062fb280-8d63-11eb-9646-5c4ca5d7d1ef.jpg)

I think it gets confusing because intelliJ by default does this:
```js
axios.post(url,{
  headers: {}
})
  .then()
  .catch()
```
instead of:
```js
axios.post(url,{
  headers: {}
})
.then()
.catch()
```